nilearn/datasets/func.py

```diff
@@ -1871,13 +1871,11 @@ def load_nki(
     ):
         logger.log(f"Loading subject {i} of {n_subjects}.", verbose=verbose)
 
-        left_data = load_surf_data(left)
-        right_data = load_surf_data(right)
         img = SurfaceImage(
             mesh=fsaverage[mesh_type],
             data={
-                "left": left_data,
-                "right": right_data,
```

Comment:
minor refactor: SurfaceImage will take care of calling load_surf_data if filenames are passed